docker/puppetdb/Dockerfile

```diff
@@ -31,7 +31,7 @@ ENV PUPPERWARE_ANALYTICS_ENABLED=false \
 # note: LOGDIR cannot be defined in the same ENV block it's used in
 # this value may be set by users, keeping in mind that some of these values are mandatory
 # -Djavax.net.debug=ssl may be particularly useful to set for debugging SSL
-ENV PUPPETDB_JAVA_ARGS="-Djava.net.preferIPv4Stack=true -Xms256m -Xmx256m -XX:+UseParallelGC -Xlog:gc*:file=$LOGDIR/puppetdb_gc.log::filecount=16,filesize=65536 -Djdk.tls.ephemeralDHKeySize=2048"
```

Comment:
This arg caused an error on startup in the JVM for "not known argument".

Reply:
Hmmm - we're successfully using that option in the PE version of PuppetDB. We should try and find the equivalent for this version of Java if it's not working here if it's a different Java runtime.

This is the code from pe-puppetdb-extensions

```
# NOTE: Docker ENV values cannot consume other ENV values, so full path to $LOGDIR is specified for -Xloggc
# this value may be set by users, keeping in mind that some of these values are mandatory
# -Djavax.net.debug=ssl may be particularly useful to set for debugging SSL
    PUPPETDB_JAVA_ARGS="-Djava.net.preferIPv4Stack=true -Xms256m -Xmx256m -XX:+UseParallelGC -Xlog:gc*:file=/opt/puppetlabs/server/data/puppetdb/logs/puppetdb_gc.log::filecount=16,filesize=65536 -Djdk.tls.ephemeralDHKeySize=2048" \
```

That first note is actually wrong about ENV vars not being able to use other ENV vars... they just have to be defined earlier like this:

```
ENV foo
ENV bar=$foo
```


/cc @puppetlabs/puppetdb

Reply:
I'll see about getting you more detailed output. My dockerfile doesn't extend this one too much.